framework/db/src/com/cloud/utils/db/Merovingian2.java

```diff
@@ -63,9 +63,7 @@
     private Merovingian2(long msId) {
         super(MerovingianMBean.class, false);
         _msId = msId;
-        Connection conn = null;
-        try {
-            conn = TransactionLegacy.getStandaloneConnectionWithException();
+        try (Connection conn = TransactionLegacy.getStandaloneConnectionWithException();) {
```

Comment:
ok : 
        Connection conn = null;
        try {
            conn = TransactionLegacy.getStandaloneConnectionWithException();
            conn.setTransactionIsolation(Connection.TRANSACTION_READ_COMMITTED);
            conn.setAutoCommit(true);
            _concierge = new ConnectionConcierge("LockMaster", conn, true);
        } catch (SQLException e) {
            s_logger.error("Unable to get a new db connection", e);
            throw new CloudRuntimeException("Unable to initialize a connection to the database for locking purposes", e);
        } finally {
            if (_concierge == null && conn != null) {
                try {
                    conn.close();
                } catch (SQLException e) {
                    s_logger.debug("closing connection failed after everything else.", e);
                }
            }